src/k8s/pkg/k8sd/features/calico/cleanup.go

```diff
@@ -0,0 +1,81 @@
+package calico
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net"
+	"os"
+	"os/exec"
+	"path"
+	"regexp"
+	"strings"
+
+	"github.com/canonical/k8s/pkg/snap"
+	"golang.org/x/sys/unix"
+)
+
+func CleanupNetwork(ctx context.Context, snap snap.Snap) error {
+	interfaces, err := net.Interfaces()
+	if err != nil {
+		return fmt.Errorf("failed to list network interfaces: %w", err)
+	}
+
+	// Find the interfaces created by Calico
+	for _, iface := range interfaces {
+		// Check if the interface name matches the regex pattern
+		match, err := regexp.MatchString("^vxlan[-v6]*.calico|cali[a-f0-9]*$", iface.Name)
```

Comment:
Why do you use a more advanced regexp for the interfaces but for iptables grepping for "cali" is sufficient?

Reply:
Also, does this catch the interfaces in case of bird (e.g. `tunl0`)?

Reply:
Calico operates on multiple modes, can create vxlan, bird and various interfaces which does not always contain a keyword. I've added `tunl` interfaces to the list.